packages/js/product-editor/src/components/schedule-sale-pricing-block/edit.tsx

```diff
@@ -0,0 +1,149 @@
+/**
+ * External dependencies
+ */
+import { DateTimePickerControl } from '@woocommerce/components';
+import { OPTIONS_STORE_NAME } from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+import { useBlockProps } from '@wordpress/block-editor';
+import { BlockEditProps } from '@wordpress/blocks';
+import { ToggleControl } from '@wordpress/components';
+import { useEntityProp } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import { createElement, useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import moment from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import { ScheduleSalePricingBlockAttributes } from './types';
+import { useValidation } from '../../hooks/use-validation';
+
+export function Edit( {}: BlockEditProps< ScheduleSalePricingBlockAttributes > ) {
+	const blockProps = useBlockProps( {
+		className: 'wp-block-woocommerce-product-schedule-sale-pricing',
+	} );
+
+	const dateTimeFormat = useSelect( ( select ) => {
+		const { getOption } = select( OPTIONS_STORE_NAME );
+		return (
+			getOption< string | null >( 'links_updated_date_format' ) ||
```

Comment:
Can we rely on the WP moment to format this correctly?

My preference would be to move this upstream if not and avoid using the options data store wherever possible.  I'm hopeful we can deprecate this in the future.

Reply:
Removed the options datastore and use `@wordpress/date` `getSettings().formats.datetime` to get the datetime format used by the `DateTimePickerControl` -> https://github.com/woocommerce/woocommerce/pull/37567/commits/83e2056c6ba236062d1c1e7966ebed4204e8260c